SMA-297: Update NYPL HTTP Android Library

robbyice opened the pull request:
**What's this do?**
This PR bumps the platform submodule to point to a new version using NYPL Http Android 0.0.25-SNAPSHOT

**Why are we doing this? (w/ JIRA link if applicable)**
Keeping dependencies up-to-date

https://jira.nypl.org/browse/SMA-297

**How should this be tested? / Do these changes have associated tests?**

Ensure the app and its associated tests run/function normally

**Dependencies for merging? Releasing to production?**
No required dependencies.

**Has the application documentation been updated for these changes?**
No documentation changes

**Did someone actually run this code to verify it works?**
I ran the app on my Pixel 2 device and bounced around a few screens without issue.

Commit:
Update NYPL HTTP Android

MalcolmMcFly commented:
@robbyice so I think what we want to do here is release a new version of NYPL HTTP Android Library first by incrementing the version and publishing 0.0.25. This should help: https://docs.google.com/document/d/1DIuMWKqpt4mNqJ1Cz795fZV5T6N3yYGyNTOY6vlMbQc/edit
Give it a look and let me know if we need to connect to walkthrough it.

MalcolmMcFly committed:
Updated platform to latest version

MalcolmMcFly committed:
Update platform

Commit:
Fix AndroidX upgrade issues

- Bump compile/target SDK version to 31
- API changes for ActivityResultContract in CardCreatorContract

Commit:
Fix DuplicateClasses issue with lifecycle-viewmodel

- Explicitly declare lifecycle-viewmodel-ktx when fragment-ktx is used to force transitive dependency to be the latest version

robbyice commented on simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/CardCreatorContract.kt:
```diff
@@ -41,9 +41,9 @@ class CardCreatorContract(
+  override fun parseResult(resultCode: Int, intent: Intent?): Output =
     if (intent == null || resultCode != Activity.RESULT_OK) {
-      null
+      Output("", "")
```
This gets the code to compile but the implications of losing nullability here probably warrants a review of where/how this contract is used.

robbyice commented:
In order to resolve the naming collision with transitive dependencies I added an explicit dependency declaration for viewmodel-ktx wherever fragment-ktx is used. Wasn't sure if this "bottom-up" approach is preferable or if we'd rather do something "top-down" - i.e., the individual applications' build.gradle file resolves this (vanilla, simplyE, openEbooks). Either way seems a little iffy.

We could also be more explicit about what we are doing by excluding viewmodel-ktx from the fragment-ktx declaration. Or do something like this: https://docs.gradle.org/current/userguide/dependency_constraints.html

Open to ideas/suggestions here.

Commit:
Minor signature changes to some ViewModel Factories

Commit:
A few more missed signature changes

Commit:
Fix some signatures for tests with Paging data

Commit:
ktLint fixes
